File: C++/1352A. Sum of Round Numbers.cpp
#include<bits/stdc++.h>
using namespace std;
#define FAST_IO ios::sync_with_stdio(false); cin.tie(0); cout.tie(0);
int main()
{
    FAST_IO;
    int t, count=0;
    cin>>t;
    for(int i=0; i<t; i++){
        int a, c, count=0;
        cin>>a;
        vector<int> b;
        for(int l=0; a>0; l++){
            c = a%10;
            a /= 10;
            if(c!=0){
                count++;
            }
            b.push_back(c);
        }
        cout<<count<<"\n";
        for(int l=b.size()-1; l>=0; l--){
            int tmp = b[l];
            if(tmp==0){
                continue;
            }
            for(int j=0; j<l; j++){
                tmp *= 10;

            }
            cout<<tmp<<" ";
        }
        cout<<"\n";
    }
    return 0;
}

File: Python/1352A. Sum of Round Numbers.py
def main():
    t = int(input())
    while t>0:
        a = int(input())
        b = []
        count = 0
        while a>0:
            c = a%10
            a = a//10
            if c!=0:
                count += 1
            b.append(c)
        print(count)
        l = len(b) -1 
        while l>=0:
            tmp = b[l]
            if tmp != 0:
                for lol in range(0, l):
                    tmp *= 10
                print(end=f"{tmp} ")
            l -= 1
        print(end="\n")
        t -= 1
main()